osu.Game/Tests/Visual/OnlinePlay/TestRoomRequestsHandler.cs

```diff
@@ -197,9 +197,8 @@ public bool HandleRequest(APIRequest request, APIUser localUser, BeatmapManager
 
                 case GetBeatmapSetRequest getBeatmapSetRequest:
                 {
-                    var baseBeatmap = getBeatmapSetRequest.Type == BeatmapSetLookupType.BeatmapId
-                        ? beatmapManager.QueryBeatmap(b => b.OnlineID == getBeatmapSetRequest.ID)
-                        : beatmapManager.QueryBeatmap(b => b.BeatmapSet.OnlineID == getBeatmapSetRequest.ID);
+                    // Incorrect logic, see https://github.com/ppy/osu/pull/28991#issuecomment-2256721076 for reason why this change
+                    var baseBeatmap = beatmapManager.QueryBeatmap(b => b.OnlineID == getBeatmapSetRequest.ID);
```

Comment:
This cannot be left in like this. If you're going to try a fix, then at least try something like

```csharp
                    var baseBeatmap = getBeatmapSetRequest.Type == BeatmapSetLookupType.BeatmapId
                        ? beatmapManager.QueryBeatmap(b => b.OnlineID == getBeatmapSetRequest.ID)
                        : beatmapManager.QueryBeatmapSet(s => s.OnlineID == getBeatmapSetRequest.ID)?.PerformRead(s => s.Beatmaps.First().Detach());
```

which at least has a semblancy of correctness and should not cause realm to die of death (although I didn't verify this, because I can't get the test failure to repro on my box, on a quick look).